core/include/vecmem/utils/deleter.hpp

```diff
@@ -38,11 +45,14 @@ namespace vecmem::details {
       void operator()( void* ptr );
 
    private:
-      /// The number of bytes allocated for the memory block
-      std::size_t m_bytes;
-      /// The memory resource used for deleting the memory block
-      memory_resource* m_resource;
+      /// The number of elements allocated for the memory block
+      size_type m_elements;
+      /// The allocator used to delete the elements
+      allocator_type m_allocator;
```

Comment:
I wasn't absolutely sure about this either. But I think just making a simple copy of the allocator received in the constructor is the way to go here. But I'm open to suggestions...

Reply:
Will `m_elements` ever be something other than 1 here?

Reply:
It's used by `vecmem::array`. (At the moment. It may see usage outside of it later on as well.) So `m_elements` is meant to be the size of the array. (Defined either at compile, or at runtime. Remember that the size of `vecmem::array` can also be defined at runtime, since it needs to use `vecmem::memory_resource` for the allocation anyway.)